crates/bevy_utils/src/intern.rs

```diff
@@ -0,0 +1,135 @@
+use std::{any::TypeId, hash::Hash};
```

Comment:
A short definition of interning would be helpful as module level docs for the casual reader.